detect_secrets/plugins/jwt.py

```diff
@@ -0,0 +1,53 @@
+"""
+This plugin finds JWT tokens
+"""
+from __future__ import absolute_import
+
+import base64
+import json
+import re
+
+from .base import RegexBasedDetector
+
+try:
+    # Python 2
+    from future_builtins import filter
+except ImportError:
+    # Python 3
+    pass
+
+
+class JwtTokenDetector(RegexBasedDetector):
+    secret_type = 'JSON Web Token'
+    denylist = [
+        re.compile(r'eyJ[A-Za-z0-9-_=]+\.[A-Za-z0-9-_=]+\.?[A-Za-z0-9-_.+/=]*'),
+    ]
+
+    def secret_generator(self, string, *args, **kwargs):
+        return filter(
+            self.is_formally_valid,
+            super(JwtTokenDetector, self).secret_generator(string, *args, **kwargs),
+        )
+
+    @staticmethod
+    def is_formally_valid(token):
+        parts = token.split('.')
```

Comment:
Noting for posterity the spec at https://tools.ietf.org/id/draft-jones-json-web-token-04.html#rfc.appendix.Appendix%20B aligns with this code

(Specifically)
```
If the length mod 4 is 0, no padding is added;
if the length mod 4 is 2, two '=' padding characters are added;
if the length mod 4 is 3, one '=' padding character is added;
if the length mod 4 is 1, the input is malformed.
```